src/libraries/System.Private.CoreLib/src/System/Globalization/SurrogateCasing.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Globalization
+{
+    internal static class SurrogateCasing
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void ToUpper(char h, char l, out char hr, out char lr)
+        {
+            Debug.Assert(char.IsHighSurrogate(h));
```

Comment:
These asserts could fail. See time-of-check/time-of-use comment in _InvariantModeCasing.cs_.

Reply:
I changed the calling site of InvariantModeCasing.CompareStringIgnoreCase so I'll keep the assert here.